pkg/controller/kubedirectorcluster/members.go

```diff
@@ -239,6 +240,45 @@ func syncMemberNotifies(
 	wgReady.Wait()
 }
 
+// https://github.com/bluek8s/kubedirector/issues/547
+// setStateDetailLogs sets the extracted results
+// of startscript executions
+// to corresponding MemberStateDetail fields
+func setStateDetailLogs(
+	readFileFn func(string, io.Writer) (bool, error),
+	stateDetail *kdv1.MemberStateDetail,
+	roleMaxLogSize string,
+) {
+
+	maxSize, _ := strconv.Atoi(roleMaxLogSize)
```

Comment:
I think there's still a couple of issues here although I could be overlooking some things.

First, I still don't see an overall hard backstop to guard against user-specified values breaking the status stanza (by being too large). To recap a couple of my earlier comments:

> I'm not sure this should be app-definable. We have limited space in the resource document (K8s API accepts up to 3 MB, etcd backend by default can store up to 1 MB) so by allowing the kdapp author to define some field size we're opening the door for them to break certain sizes of kdcluster.
> 
> So I guess my opinion is that KD should always have some hard character limit for these fields, regardless of anything else.

Second, it looks like it's possible for roleMaxLogSize to remain undefined. In the current way things are typed, that means empty-string. That would cause an error here on this line.

I think both of these issues could be solved by having a hard limit imposed by KD, for example 256 chars. The hard limit would come into play if roleMaxLogSize is undefined or if it is larger than the hard limit.

I'm still not entirely convinced of the utility of letting the kdapp developer specify some limit here, but in any case we do need the hard limit as a backstop.